src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/BookingEntity.kt

```diff
@@ -13,7 +14,13 @@ import javax.persistence.OneToOne
 import javax.persistence.Table
 
 @Repository
-interface BookingRepository : JpaRepository<BookingEntity, UUID>
+interface BookingRepository : JpaRepository<BookingEntity, UUID> {
+  @Query("SELECT b FROM BookingEntity b WHERE b.premises.id = :premisesId AND b.arrivalDate <= :endDate AND b.departureDate >= :startDate")
+  fun findAllByPremisesIdAndOverlappingDate(premisesId: UUID, startDate: LocalDate, endDate: LocalDate): MutableList<BookingEntity>
```

Comment:
just curious, why does this have to return a `MutableList` ?

Reply:
Would seem it doesn't, have switched to List